lens_timer/Public/ExpressionController_Bilateral.js

```diff
@@ -0,0 +1,212 @@
+// -----JS CODE-----
+// @input Component.FaceMaskVisual target
+// @input Component.RenderMeshVisual faceMesh
+
+// @input string expressionRight
+// @input string expressionLeft
+// @input string displayText
+// @input string finishText
+// @input number completedReps
+// @input number requiredReps
+// @input number minExpressionValue
+// @input number expressionIndex
+
+const pubSub = require("./PubSubModule");
+//let minExpressionValue = global.ExpressionMinValues[script.expressionRight];
+var midRep;
+var color;
+var sensitivity;
+var isRightDetectionOn;
+var isLeftDetectionOn;
+global.timerUpdate = 0;
+// face mask visual disabled by default
+script.target.enabled = false;
+
+/***
+* Called once when onAwake
+*/
+function Initialize() {
+// Set initial values
+    midRep = false;
+    color = script.target.getMaterial(0).getPass(0).baseColor;
+    sensitivity = global.Sensitivity;
+    // Display prompt text
+    pubSub.publish(pubSub.EVENTS.SetExpressionRequiredRepText,  script.requiredReps.toString());
+    pubSub.publish(pubSub.EVENTS.SetExpressionPromptText, script.displayText);
+    SetBilateralDetection();
+    SetEvents();
+}
+
+/***
+* Set functions to be called every frame
+*/
+function SetEvents() {
+  global.timerUpdate = 0;
+  global.complete = 0;
+  var updateEvent = script.createEvent("UpdateEvent");
+  updateEvent.bind(OnUpdate);
+}
+
+/***
+* Things to be called every frame
+*/
+function OnUpdate(){
+  // todo is sensitivity a per exercise or global thing?
+  
+  sensitivity = global.Sensitivity;
+  CountReps();
+  UpdateVisual(script.target);
+}
+
+/***
+* Update opacity of mask based on expression weight
+*/
+function UpdateVisual(visualComponent) {
+     alpha = GetAdjustedWeight();
+     color = visualComponent.getMaterial(0).getPass(0).baseColor;
+     visualComponent.getMaterial(0).getPass(0).baseColor = new vec4(color.r, color.g, color.b, alpha);
+ }
+
+// TODO: Still trying to figure out how to count reps when a persons base line values may differ significantly
+/***
+* Count completed reps, expression must return to base line bf another rep is counted.
+*/
+function CountReps() {
+
+    //   print(global.complete);
+  // stop counting when hit required reps
+  if (global.complete == 1){
+    Finished();
+    return;
+  }
+    // Update rep count text
+     pubSub.publish(pubSub.EVENTS.SetExpressionRepText,  script.completedReps.toString());
+
+     var adjustedWeight = GetAdjustedWeight();
+     //print("adjusted " + adjustedWeight)
+    
+  //  print("adj - minexprvalue");
+  //  print(adjustedWeight - script.minExpressionValue);    
+     if (adjustedWeight > script.minExpressionValue && midRep !== true){
+        midRep = true;
+        script.completedReps += 1;                
+        global.timerUpdate = 1;
```

Comment:
I'd love for the timer stuff to be in a separate function and we can have two bools isReps or isTimed that determine if the exercise should count reps or count down the time.

I am also worried if the script has been changed since you worked on it. I wanna keep as many scripts the same as possible. So if we can make sure this script and the one in the working copy end up the same that would be awesome.